Standards/scs-XXXX-vN-mandatory-and-supported-IaaS-services.md

```diff
@@ -0,0 +1,80 @@
+For this purpose, this standard will establish lists for mandatory services that have to be present in a SCS cloud as well as supported services, which are considered by some standards and may be tested or even implemented in the reference implementation but are optional in a sense that their omission will not violate SCS conformance.
+
+## Motivation
+
+There are many OpenStack APIs and their corresponding services that can be deployed on the IaaS level.
+These services have differences in the quality of their implementation and liveness and some of them may be easily omitted when creating an IaaS deployment.
+To fulfill all SCS-provided standards only a subset of these APIs are required.
+Some more but not all remaining OpenStack APIs are also supported additionally by the SCS project and may be part of its reference implementation.
+This results in different levels of support for specific services.
+This document will give readers insight about how the SCS classifies the OpenStack services accordingly.
+If a cloud provides all mandatory and any number of supported OpenStack APIs, it can be tested for SCS-compliance.
+Any unsupported services will not be tested.
+
+## Mandatory OpenStack services
+
+The following OpenStack services MUST be present in SCS-compliant IaaS deployments:
+
+| OpenStack Service | description |
+|-----|-----|
+| **Cinder** | Block Storage service |
+| **Glance** | Image service |
+| **Keystone** | Identity service |
+| **Neutron** | Networking service |
+| **Nova** | Compute service |
+| **Octavia** | Load-balancer service |
+| **Placement** | Hardware Describing Service for Nova |
+| **S3 API object storage** | No formal standard exists, many implementations: Swift, RadosGW, minio... |
+
+:::caution
+
+S3 API implementations may differ in certain offered features.
+CSPs must publicly describe, which implementation they use in their deployment.
+Users should always research whether a needed feature is supported in the offered implementation.
+
+:::
+
+## Supported OpenStack services
+
+The following services MAY be present in SCS-compliant IaaS deployment and are considered in the SCS standards.
+
+| OpenStack Service | description |
+|-----|-----|
+| **Barbican** | Key Manager service |
+| **Cloudkitty** | Rating/Billing service |
+| **Ceilometer** | Telemetry service |
+| **Designate** | DNS service |
+| **Gnocchi** | Time Series Database service |
+| **Heat** | Orchestration service |
+| **Horizon** | Dashboard |
+| **Ironic** | Bare Metal provisioning service |
+| **Manila** | Shared File Systems service |
+| **Masakari** | Instances High Availability service |
+| **Skyline** | Dashboard |
+
+## Unsupported OpenStack services
+
+All other OpenStack services that are not mentioned in the mandatory or supported lists are not tested for their compatibility and behavior in SCS clouds by the SCS community.
```

Comment:
This wording could be read to imply that all supported services are tested for their compatibility and behavior, which I don't think is currently the case. If this is the case and I'm just unaware of the tests being done, please point me to these tests.

Also I'm not sure what is referred to when talking about "tested for their [..] behavior"? Does it mean the behavior of the service is SCS compliant - whatever that means, given that I'm not aware of any service specific standards with regards to the list of "supported" services? Maybe this sentence could be clarified a bit?

The only somewhat service specific standards I can think off of the top of my head are the image naming standard which could be considered to be glance specific and the entropy standard which could be viewed as nova specific, if you squint. So if there are service specific standards which we test against we should maybe list those here or at least link to them?

Reply:
In another comment @berendt mentioned that, except for Masakari, all supported services are available in the reference implementation. As I read his comment, they will eventually also test Masakari.

In deed there are standards that will and have to be applied to all used services: e.g. the role standard that is currently under development. This is also what I meant with behavior: if services have different role names or have a different meaning of role names, their behavior is different.
But i should rephrase this sentece, for better understanding.

Reply:
Please don't conflate availability of services with "tested" services.
Not all services that are available to use are necessarily also tested.

Also I put tested in quotation marks, because it's really not defined anywhere what that actually means, beside SCS standard tests.

Reply:
marking as resolved as we mostly talk about APIs now.